data_resource_api/factories/orm_factory.py

```diff
@@ -65,43 +65,74 @@ def evaluate_foreign_key(self, foreign_keys, field_name, field_type):
                 return True, foreign_key_reference
         return False, None
 
-    def create_sqlalchemy_fields(self, fields: dict, primary_key, foreign_keys=[]):
+    def create_sqlalchemy_fields(self, descriptor_fields: list, primary_key: str, foreign_keys=[]):
         """Build SQLAlchemy fields to be added to new table object.
 
         Args:
-            fields (dict):
+            descriptor_fields (list of dict):
             primary_key (str): The primary key field.
             foreign_keys (list): Collection of foreign key fields.
 
         Return:
             dict: SQLAlchemy fields to append to the new database object.
 
         """
+
+        def is_required(descriptor_field):
+            has_required_key_and_true = ('required' in descriptor_field.keys() and descriptor_field['required']) 
+            has_required_in_constraints = (
+                'constraints' in descriptor_field.keys() 
+                and 'required' in descriptor_field['constraints'].keys() 
+                and descriptor_field['constraints']['required']
+            )
+            return has_required_key_and_true or has_required_in_constraints
+
         sqlalchemy_fields = {}
+
         if isinstance(primary_key, str):
             primary_key = [primary_key]
-        for field in fields:
-            if ('required' in field.keys() and field['required']) or \
-                ('constraints' in field.keys() and 'required' in field['constraints'].keys() and field['constraints']['required']):
+
+        for descriptor_field in descriptor_fields:
+            if is_required(descriptor_field):
                 nullable = False
             else:
                 nullable = True
-            if field['name'] in primary_key:
-                sqlalchemy_fields[field['name']] = Column(
-                    self.get_sqlalchemy_type(field['type']), primary_key=True)
-            else:
-                is_foreign_key, reference_table = self.evaluate_foreign_key(
-                    foreign_keys, field['name'], field['type'])
-                if not is_foreign_key:
-                    sqlalchemy_fields[field['name']] = Column(
-                        self.get_sqlalchemy_type(field['type']), nullable=nullable)
-                else:
-                    try:
-                        sqlalchemy_fields[field['name']] = Column(
-                            self.get_sqlalchemy_type(
-                                field['type']), ForeignKey(reference_table, onupdate='CASCADE', ondelete='CASCADE'))
-                    except Exception as e:
-                        print('An exception occured {}'.format(e))
+            
+            # If this is the primary key
```

Comment:
This lingering comments (lines 101, 109, and 130) do not add much to the code. I'd say either: consolidate them in the docstring for `create_sqlalchemy_fields`, or just omit them entirely.